src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerationTemplate.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public abstract class LinkGenerationTemplate
+    {
+        public abstract string MakeUrl(object values);
```

Comment:
Does this need to be abstract? IMO just have a non-virtual method call MakeUrl with null LinkOptions.

Reply:
I agree. Changed it now.